server/src/main/java/org/apache/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -813,6 +851,24 @@ private DataSegment mergeAndPush(
       Closer closer = Closer.create();
       try {
         for (FireHydrant fireHydrant : sink) {
+
+          // if batch, swap/persist did not memory map the incremental index, we need it mapped now:
+          if (!isRealTime()) {
+            // sanity:
+            if (fireHydrant.getPersistedFile() == null) {
+              throw new ISE("Persisted file for batch hydrant is null!");
```

Comment:
would you want any other info in this message to debug the exception? It's ok if not.

Reply:
That should never happen... it is just sanity. Can you think of anything else to say?

Reply:
Added hydrant (toString), a little more but not a lot more.